### PR TITLE
[chrome] Add partitionKey to chrome.cookies

### PR DESCRIPTION
**Pull request by amakhrov**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.chrome.com/docs/extensions/reference/api/cookies#partitioning
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/chrome/test/index.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 14ebcad55e7e61f58adfed50483b637e0c1c2e00
</p>

File: types/chrome/test/index.ts [hunk 1]
```diff
@@ -1853,6 +1853,16 @@ async function testBrowsingDataForPromise() {
     await chrome.browsingData.removeIndexedDB({});
 }
 
+// https://developer.chrome.com/docs/extensions/reference/cookies
+async function testCoookies() {
```
Comment (erwanjugand) [inline types/chrome/test/index.ts:1857]: Maybe merge `testCoookies` into `testCookieForPromise` :)

Comment (amakhrov) [reply types/chrome/test/index.ts:1857]: 🤦‍♂️
I swear I have looked for existing cookies tests in this file

File: types/chrome/index.d.ts [hunk 2]
```diff
@@ -1822,6 +1822,8 @@ declare namespace chrome.cookies {
         domain: string;
         /** The name of the cookie. */
         name: string;
+        /** The partition key for reading or modifying cookies with the Partitioned attribute. */
```
[the remark this answers is not in the document]
Comment (amakhrov) [reply types/chrome/index.d.ts]: done

File: types/chrome/test/index.ts (hunk 1 above)
Comment (amakhrov) [reply types/chrome/test/index.ts]: done

**Review by erwanjugand — APPROVED** [review]
Looks good to me

File: types/chrome/index.d.ts [hunk 3]
```diff
@@ -1845,6 +1850,14 @@ declare namespace chrome.cookies {
         sameSite: SameSiteStatus;
     }
 
+    /** Represents a partitioned cookie's partition key. */
+    export interface CookiePartitionKey {
+        /** Indicates if the cookie was set in a cross-cross site context. This prevents a top-level site embedded in a cross-site context from accessing cookies set by the top-level site in a same-site context. */
+        hasCrossSiteAncestor?: boolean;
+        /** The top-level site the partitioned cookie is available in. */
+        topLevelSite?: string;
```
Comment (nicolas377) [inline types/chrome/index.d.ts:1858]: In general, the style in DT prefers optional types to explicitly declare `T | undefined`.
```suggestion
        topLevelSite?: string | undefined;
```

File: types/chrome/index.d.ts [hunk 4]
```diff
@@ -1845,6 +1850,14 @@ declare namespace chrome.cookies {
         sameSite: SameSiteStatus;
     }
 
+    /** Represents a partitioned cookie's partition key. */
+    export interface CookiePartitionKey {
+        /** Indicates if the cookie was set in a cross-cross site context. This prevents a top-level site embedded in a cross-site context from accessing cookies set by the top-level site in a same-site context. */
+        hasCrossSiteAncestor?: boolean;
```
Comment (nicolas377) [inline types/chrome/index.d.ts:1856]: The docs say this property is pending (not yet in a stable release). IMO, this shouldn't be in the types if it isn't in the actual API yet.

File: types/chrome/index.d.ts [hunk 5]
```diff
@@ -1898,6 +1921,11 @@ declare namespace chrome.cookies {
 
     export interface Details {
```
Comment (nicolas377) [inline types/chrome/index.d.ts:1922]: You don't need to do this, but the docs call this interface `CookieDetails`. Would you mind changing that name and throwing some docs on it?

**Review by nicolas377 — CHANGES_REQUESTED** [review]

File: types/chrome/index.d.ts (hunk 5 above)
Comment (amakhrov) [reply types/chrome/index.d.ts:1922]: This type is exported. Would not it be a breaking change if I rename it? How do you approach breaking changes in DT?

File: types/chrome/index.d.ts (hunk 3 above)
Comment (amakhrov) [reply types/chrome/index.d.ts]: done

File: types/chrome/index.d.ts (hunk 4 above)
Comment (amakhrov) [reply types/chrome/index.d.ts]: done

File: types/chrome/index.d.ts (hunk 5 above)
Comment (amakhrov) [reply types/chrome/index.d.ts:1921]: I added the docs. Will be happy to rename to `CookieDetails`, too, once you confirm it's ok to include a potentially breaking change in this PR

Comment (nicolas377) [reply types/chrome/index.d.ts:1921]: Unfortunately, chrome types are a bit of a weirder thing in DT. Normally, `@types` follows the package versioning (generally semver) until patches (which are incremented on each release of the types package). The thing with chrome is that there is no real versioning on chrome API's. You could argue that Chrome versions are version enough to follow them, but you need to remember that extensions have to run across multiple major versions of chrome. Also, you need to consider that chrome extensions can run on all chromium based browsers (Chrome, Edge, Brave, ect.). Our solution is to forgo versioning, and just call everything 0.0.x (I think we're closing in on 300 now). It's annoying but essentially breaking changes happen all the time when chrome API's are updated. Chrome is good with giving backwards compatibility and letting developers update their code as API's come into/out of existence, so we have a bit of built in backwards compatibility. In this case, we made a mistake naming something (or the name was changed), so we should 100% fix that, and there's no precedent for versioning breaking changes in this world, so you're all good to go.

TL;DR: Yes, it's fine to include a breaking change as long as it follows the chrome API's.

Comment (amakhrov) [reply types/chrome/index.d.ts]: Got it, thanks for the context!

It's done, too

Comment (erwanjugand) [reply types/chrome/index.d.ts]: Hello @nixolas377 just a quick question ☺️

Since we're not tied to an API version, would it be possible to bump the package version to 0.1.999 to indicate a breaking change ?

**Review by sandersn — APPROVED** [review]

**Comment by amakhrov** [comment]
Ready to merge

File: types/chrome/index.d.ts (hunk 5 above)
Comment (nicolas377) [reply types/chrome/index.d.ts]: I wouldn't see a reason why, to me I think the reason we do this versioning is to get rid of any confusion with a version of Chrome. These types are for the latest version of chrome (as discussed above they have backwards compatibility built in), and since there's so many chromium browsers with the same/extremely similar apis, all with competing versioning, we don't want to confuse anyone.
